### PR TITLE
Resharding V3 - add a few state sync details

### PR DESCRIPTION
**Pull request by marcelo-gonzalez**

### CODE REVIEW

File: neps/nep-0568.md [hunk 1]
```diff
@@ -135,6 +135,13 @@ supporting smooth transitions without altering storage structures directly.
...
+
+When nodes sync state (either because they've fallen far behind the chain, or because they're going to become a chunk producer for a new shard in a future epoch), they first identify a point in the chain they'd like to sync to. This is always the first block of the current epoch, which the node should be aware of once it has synced headers to the current point in the chain. The hash of this first block is referred to as the "sync_hash" in many places in the state sync implementation. Then the node makes a request (currently to centralized storage on GCS, but in the future to other nodes in the network) for a `ShardStateSyncResponseHeader` corresponding to that "sync_hash" and
+the Shard ID of the shard it's interested in. Among other things, this header includes the last new chunk before "sync_hash" in the shard, and a `StateRootNode` with hash equal to that chunk's `prev_state_root` field. Then the node downloads (again from GCS, but in the future it'll be from other nodes) the nodes of the trie with that `StateRootNode` as its root. Afterwards, it applies new chunks in the shard until it's caught up.
```
Comment (wacban) [inline neps/nep-0568.md:141]: This section may be a bit too technical for the specification. Perhaps you can move the implementation details to the "reference implementation" section below and here keep it at a higher level?

Comment (marcelo-gonzalez) [reply neps/nep-0568.md]: yea good point, done

File: neps/nep-0568.md [hunk 2]
```diff
@@ -252,6 +251,12 @@ For archival nodes, mappings are retained permanently to ensure access to the hi
...
+
+ As described above, the state we download is the state in the shard after applying the second to last new chunk before `sync_hash`, which belongs to the previous epoch (since `sync_hash` is the first block of the new epoch). To move the point in the chain of the initial state download to the current epoch, we could either move the `sync_hash` forward or we could change the state sync protocol (perhaps changing the meaning of the `sync_hash` and the fields of the `ShardStateSyncResponseHeader`, or somehow changing these structures more significantly). The former is an easier first implementation, since it would not require any changes to the state sync protocol other than to the expected `sync_hash`. We would just need to move the `sync_hash` to a point far enough along in the chain so that the `StateRootNode` in the `ShardStateSyncResponseHeader` refers to state in the current epoch.
```
Comment (wacban) [inline neps/nep-0568.md:258]: Can you clarify which one do we want to do?

**Review by wacban — APPROVED** [review]
LGTM

After merging into the main PR please see if there are any lint errors and fix those. For some reason those only show up on the main RP.

**Pull request merged**